controller/tenant.go

```diff
@@ -83,12 +83,15 @@ func (c *TenantController) Setup(ctx *app.SetupTenantContext) error {
 		return jsonapi.JSONErrorResponse(ctx, errors.NewInternalError(ctx, err))
 	}
 
+	username := tenant.ConstructNsUsername(c.tenantService, env.RetrieveUserName(user.OpenShiftUsername))
+
 	// create openshift config
 	openshiftConfig := openshift.NewConfig(c.config, user.UserData, cluster.User, cluster.Token, cluster.APIURL)
 	tenant := &tenant.Tenant{
 		ID:         ttoken.Subject(),
 		Email:      ttoken.Email(),
 		OSUsername: user.OpenShiftUsername,
+		NsUsername: username,
```

Comment:
I called it `ns_username` because there in the templates it is used as: 
```yaml
namespace: ${USER_NAME}-che
``` 
so, I wanted to make it consitent. Unfortunatelly, the fact that we also have OSUsername that is stored in the table as `username` make the stuff more complicated. 

> So, I would like to make it clear across the code that we have clear names for these two concepts here. Username (actual username) and a base-name or prefix which is currently constructed from a username and an index.

The actual `username` is/was used only for the namespaces - there is no other usage of it (if I'm not mistaken). We only use OSUsername. In other words, the `username` is only our artificial thingy (implementation detail) that is retrieved from the real OSUsername. Unfortunately, we used this name of the variable also in the templates - which makes sense in case of user namespace.

So, I'll call it `NsBaseName` and when this PR is merged I'll change also the variables in all templates from `${USER_NAME}` to `${NS_BASE_NAME}` to make it consistent, if you are fine with it.

Why is naming always so hard? :-/

Reply:
“There are only two hard things in Computer Science: cache invalidation and naming things.”
:)